teps/0036-start-measuring-tekton-pipelines-performance.md

```diff
@@ -0,0 +1,99 @@
+---
+status: proposed
+title: Start Measuring Tekton Pipelines Performance
+creation-date: '2020-11-20'
+last-updated: '2020-11-20'
+authors:
+- '@bobcatfish'
+---
+
+# TEP-0036: Start Measuring Tekton Pipelines Performance
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Up until this point, we have left it to our users to report back to us on how well Tekton Pipelines performs. Relying on
+our users in this way means that we can’t easily inform users about what performance to expect and also means that we
+usually only catch and fix performance issues after releases.
+
+This proposal is all about finding an incremental step forward we can take towards being more proactive in responding to
+performance issues, and will also enable us to be able to evaluate the performance impact of implementation decisions
+going forward.
+
+Issue: [tektoncd/pipeline#540](https://github.com/tektoncd/pipeline/issues/540)
+
+## Motivation
+
+* To be able to understand and communicate about Tekton performance on an ongoing basis
+* Be able to answer questions like:
+  * How much overhead does a Task add?
```

Comment:
> "How much overhead does a Task add to Pod performance?"

I think pod performance depends on the cloud provider or on the environment that its getting created in. I believe whatever benchmarking we put in place will be against our own CI system to begin with, please correct me if I am wrong. I think what we are trying to evaluate here is "How much overhead these changes or this release add or improve to creating a taskrun" which includes setting up resources or workspaces or volumes etc before actually a pod is created.

Reply:
Great point @sbwsg ! I've added a bit more detail, still really vague tho so lemme know if you want me to be more precise.

@pritidesai it's true the the performance depends on the cloud provider - specifically what we're interested in here is the relative performance, and @sbwsg pointed out that half the statement was missing XD

The idea is: If i run some specific image in a pod, it takes X length of time to execute (probably on average but maybe we'll be more interested in some other way of looking at the metric over time), how much overhead does it add if I execute that same image / pod via a Task? And what if I put that Task in a Pipeline? You're right that the overhead might be different in different environments, it could be interesting to compare.